mail_activity_filter_internal_user/models/mail_activity.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2022 Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class MailActivity(models.Model):
+
+    _inherit = "mail.activity"
+
+    filter_internal_user = fields.Boolean(
+        string="Assign to internal user only",
+        default=True,
+    )
+
+    @api.onchange("filter_internal_user")
+    def onchange_filter_internal_user(self):
+        """
+        Change the domain on the user_id field based on the value of
+        filter_internal_user.
+        """
+        if "user_id" in self._fields:
+            filter_internal_user_domain = ("share", "=", False)
+            previous_domain = self._fields["user_id"].domain
+            if self.filter_internal_user:
+                if filter_internal_user_domain not in previous_domain:
+                    previous_domain.append(filter_internal_user_domain)
+            else:
+                previous_domain.remove(filter_internal_user_domain)
+            return {"domain": {"user_id": previous_domain}}
```

Comment:
i think this is very dangerous: instead of returning a new domain, the internal domain is modified. this causes the domain to change for all databases, because `_fields` is defined on the class. to change the domain, the internal one can be copied (with `.copy()` for example) and the copy changed afterward, but the original one must not be modified.

Reply:
if it is required to be able to choose whether to filter on internal users or not, then i think this code would do:

```suggestion
        # copy to ensure original domain is not modified
        user_domain = self._fields["user_id"].domain.copy()
        if self.filter_internal_user:
            user_domain.append(("share", "=", False))
        return {"domain": {"user_id": user_domain}}
```

instead of `.copy()` and `.append()`, we could also do:
```python
            # not .append() or += because the original domain must not be modified.
            user_domain = user_domain + [("share", "=", False)]
```
but explicit is better than implicit, and also, even if it is not changed, returning the original domain variable should be avoided, as we don’t know what will happen to it.

also, maybe there is a cleaner way to access the current domain than accessing `self._fields` directly? i don’t know.